src/custom/hooks/useApproveCallback/index.ts

```diff
@@ -29,22 +41,57 @@ export function useApproveCallbackFromTrade(
 
   const vaultRelayer = chainId ? GP_VAULT_RELAYER[chainId] : undefined
 
-  return useApproveCallback(openTransactionConfirmationModal, closeModals, amountToApprove, vaultRelayer)
+  return useApproveCallback({
+    openTransactionConfirmationModal,
+    closeModals,
+    amountToApprove,
+    spender: vaultRelayer,
+    amountToCheckAgainstAllowance,
+  })
 }
 
 export type OptionalApproveCallbackParams = {
   transactionSummary: string
 }
 
-export function useApproveCallbackFromClaim(
-  openTransactionConfirmationModal: (message: string) => void,
-  closeModals: () => void,
-  amountToApprove: CurrencyAmount<Currency> | undefined
-) {
+type ApproveCallbackFromClaimParams = Omit<
+  ApproveCallbackParams,
+  'spender' | 'amountToApprove' | 'amountToCheckAgainstAllowance'
+> & {
+  claim: EnhancedUserClaimData
+  investmentAmount?: CurrencyAmount<Currency>
+}
+
+export function useApproveCallbackFromClaim({
+  openTransactionConfirmationModal,
+  closeModals,
+  claim,
+  investmentAmount,
+}: ApproveCallbackFromClaimParams) {
   const { chainId } = useActiveWeb3React()
+  const supportedChain = supportedChainId(chainId)
 
   const vCowContract = chainId ? V_COW_CONTRACT_ADDRESS[chainId] : undefined
 
+  // Claim only approves GNO and USDC (GnoOption & Investor, respectively.)
+  const approveAmounts = useMemo(() => {
+    if (supportedChain && (claim.type === ClaimType.GnoOption || claim.type === ClaimType.Investor)) {
```

Comment:
I guess this can be more general if instead of using a `||` of all the currencies you can approve, you check if `currency` `isNative`